### PR TITLE
[MM-50475][MM-50775] - Post component fixes

### PR DESCRIPTION
**Pull request by nevyangelova**
#### Summary
This PR addresses bug fixes related to the post component refactor. One is a class mismatch on post threads in compact message view, and the other is adding back the channel header name for all RHS screens.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-50775
https://mattermost.atlassian.net/browse/MM-50475

#### Release Note
```release-note
Post component bug fixes
```

### CODE REVIEW

**Comment by amyblais** [comment]
@AshishDhama @koox00 @furqanmlk Please help prioritize this PR review.

File: package-lock.json
Comment (koox00) [inline package-lock.json:8]: I'm always confused on this change, do we need that or we should remove it?

**Review by koox00 — APPROVED** [review]

**Review by AshishDhama — APPROVED** [review]

File: package-lock.json
Comment (AshishDhama) [reply package-lock.json:8]: I think we need that. The node version I am on right now adds it automatically, but the one I was earlier was removing it.

File: components/post/post_component.tsx
Comment (AshishDhama) [inline components/post/post_component.tsx:484]: @nevyangelova About this chagne, Asaad told use that we could leave in case of Pinned post https://community.mattermost.com/core/pl/uegre6shktf8zpzzu8zce3hiea

Comment (AshishDhama) [reply components/post/post_component.tsx:484]: cc// @asaadmahmood

Comment (nevyangelova) [reply components/post/post_component.tsx:484]: @AshishDhama I have spinned this test server with 7.1 version and the pinned posts should have the headline too 
https://nevytest.test.mattermost.cloud/lalalalalalal/channels/town-square

Comment (AshishDhama) [reply components/post/post_component.tsx:484]: Yeah check the thread we had that discussion

Comment (nevyangelova) [reply components/post/post_component.tsx:484]: ok I will remove it

Comment (AshishDhama) [reply components/post/post_component.tsx:484]: But I leave that to asaad, and to make it more clear what you have here is exactly like it was before common post component.

Comment (nevyangelova) [reply components/post/post_component.tsx:484]: Still it seems per discussion that this is a redundant information, removing for now as to not block the release fix and we can add it later if we decide :)

**Review by furqanmlk — COMMENTED** [review]
Everything is working fine except the following issue:
Team name and Channel name also show up in the message area.

image

**Comment by nevyangelova** [comment]
@furqanmlk fixed

**Comment by nevyangelova** [comment]
e2e-test looks better than master. I will merge this as we wanna solve the release bugs

**Comment by amyblais** [comment]
@nevyangelova Please help open a manual cherry-pick PR for the ``release-7.9`` branch.

**Pull request merged**
